js/siteorigin-panels/model/builder.js

```diff
@@ -292,4 +292,174 @@ module.exports = Backbone.Model.extend({
 			   position === this.layoutPosition.AFTER ||
 			   position === this.layoutPosition.REPLACE;
 	},
+
+	/**
+	 * Convert HTML into Panels Data
+	 * @param html
+	 */
+	getPanelsDataFromHtml: function( html, editorClass ){
+		var thisModel = this;
+		var $html = jQuery( '<div id="wrapper">' + html + '</div>' );
+
+		if( $html.find('.panel-layout .panel-grid').length ) {
+			// This looks like Page Builder html, lets try parse it
+			var panels_data = {
+				grids: [],
+				grid_cells: [],
+				widgets: [],
+			};
+
+			// The Regex object that'll match SiteOrigin widgets
+			// console.log( panelsOptions.siteoriginWidgetRegex );
+			var re = new RegExp( panelsOptions.siteoriginWidgetRegex , "i" );
+
+			$html.find('> .panel-layout > .panel-grid').each( function( ri, el ){
```

Comment:
Would be nice if this could be handled by separate functions, e.g. `parseRow`, `parseCell`, `parseWidget` etc to try reduce the size of this function and make it more readable. To keep the functions bound to the builder model, you can use underscore's [bindAll](http://underscorejs.org/#bindAll) e.g. `_.bindAll(this, 'parseRow', 'parseCell', 'parseWidget');` in `initialize()`.